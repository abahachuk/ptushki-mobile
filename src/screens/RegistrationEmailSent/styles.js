import { StyleSheet } from "react-native";

/* eslint  import/prefer-default-export: 0 */
export const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "space-between"
  },
  logoImg: {
    width: 300,
    height: 280,
    alignSelf: "center"
  },
  headerText: {
    fontSize: 20,
    textAlign: "center"
  },
  statusText: {
    fontWeight: "bold",
    textAlign: "center",
    marginLeft: 6,
    marginTop: 16,
    fontFamily: "Roboto",
    fontSize: 16,
    lineHeight: 24
  },
  hintText: {
    textAlign: "center",
    marginLeft: 6,
    marginBottom: 16,
    color: "#546E7A",
    fontFamily: "Roboto",
    fontSize: 14,
    lineHeight: 24
  }
});
