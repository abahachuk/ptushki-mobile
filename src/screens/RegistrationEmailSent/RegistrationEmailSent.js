import React from "react";
import PropTypes from "prop-types";
import {
  View,
  Text,
  Image,
  KeyboardAvoidingView,
  ScrollView
} from "react-native";

import { Button } from "../../components";
import { translate } from "../../i18n";
import { styles } from "./styles";
/* eslint react/jsx-one-expression-per-line: 0 */

const logoImg = require("../../assets/logotype/logotype2x.png");

const RegistrationEmailSent = props => {
  const { navigation } = props;

  const goToLogin = () => {
    navigation.navigate("login");
  };

  const origin = navigation.getParam("origin", "registration");

  return (
    <ScrollView showsVerticalScrollIndicator={false}>
      <KeyboardAvoidingView style={styles.container} behavior="padding" enabled>
        <View>
          <Image style={styles.logoImg} resizeMode="contain" source={logoImg} />
          <Text style={styles.headerText}>
            {translate("login.bandingCenter")}
          </Text>
          <Text style={styles.statusText}>
            {translate(`${origin}.statusText`)}
          </Text>
          <Text style={styles.hintText}>{translate(`${origin}.hintText`)}</Text>
        </View>
        <View>
          <Button
            caption={translate("registrationEmailSent.goToLogin")}
            onPress={goToLogin}
            appearance="Dark"
          />
        </View>
      </KeyboardAvoidingView>
    </ScrollView>
  );
};

RegistrationEmailSent.navigationOptions = {
  title: translate("login.sign-in"),
  header: null
};
RegistrationEmailSent.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
    goBack: PropTypes.func
  }).isRequired
};

RegistrationEmailSent.defaultProps = {};

export default RegistrationEmailSent;
