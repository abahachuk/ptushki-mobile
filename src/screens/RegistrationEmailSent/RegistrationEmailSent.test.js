import React from "react";
import renderer from "react-test-renderer";

import RegistrationEmailSent from "./RegistrationEmailSent";
/* eslint  no-undef: 0 */
describe("Registration screen", () => {
  test("renders correctly", () => {
    const navigation = {
      navigate: jest.fn(),
      getParam: () => "succes"
    };
    const tree = renderer
      .create(<RegistrationEmailSent navigation={navigation} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
