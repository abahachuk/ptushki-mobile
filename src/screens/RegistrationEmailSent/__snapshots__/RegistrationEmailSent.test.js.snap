// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration screen renders correctly 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View>
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/logotype/logotype2x.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "height": 280,
              "width": 300,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          [missing "en.login.bandingCenter" translation]
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Roboto",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "marginLeft": 6,
              "marginTop": 16,
              "textAlign": "center",
            }
          }
        >
          succes
        </Text>
        <Text
          style={
            Object {
              "color": "#546E7A",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "lineHeight": 24,
              "marginBottom": 16,
              "marginLeft": 6,
              "textAlign": "center",
            }
          }
        >
          succes
        </Text>
      </View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 6,
                "paddingBottom": 13,
                "paddingTop": 13,
                "width": "100%",
              },
              Object {
                "backgroundColor": "#546e7a",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#546e7a",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "textAlign": "center",
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                },
              ]
            }
          >
            [missing "en.registrationEmailSent.goToLogin" translation]
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
